static/js/services/playsound.js

```diff
@@ -156,7 +156,65 @@ define(['underscore', 'Howler', 'require'], function(_, Howler, require) {
 	window.PLAYSOUND = registry; // make available for debug.
 
 	// playSound
-	return [function() {
+	return ['translation', 'appData', 'mediaStream', function(translation, appData, mediaStream) {
+
+		/*
+		  Api
+		  0: Default (Play sound)
+		  1: No sounds when in call
+		  2: No sounds
+		*/
+		var scope;
+		var playSoundDefaults = {
+			join: 0,
+			chat: 0,
+			call: 0
+		}
+		var playSound = {};
+		var inCall = false;
+		var playSoundOptions = [
+			translation._("Default"),
+			translation._("No sounds when in call"),
+			translation._("No sounds")
+		];
+		var shouldPlaySound = function(id) {
+			var play = true;
+
+			if (inCall && playSound.join === 1 || playSound.join === 2) {
+				if (id === "left" || id === "joined") {
+					play = false;
+				}
+			}
+			if (inCall && playSound.chat === 1 || playSound.chat === 2) {
+				if (id === "announce") {
+					play = false;
+				}
+			}
+			if (inCall && playSound.call === 1 || playSound.call === 2) {
+				if (id === "ring" || id === "end" || id === "dial" || id === "connect" || id === "prompt") {
+					play = false;
+				}
+			}
+
```

Comment:
Whoops, didn't read completely - you are checking different properties of `playSound`, so no change required.